test/scraping.js

```diff
@@ -139,7 +139,37 @@ describe('scraping', function() {
 			.then(function(results) {
 				var authors = results.authors;
 				assert.deepEqual(JSON.stringify(authors), expectedAuthors);
-			})
+			});
+		});
+	});
+
+	it('should get basic EPrints metadata', function() {
+		url = 'http://eprints.gla.ac.uk/113711/';
+		return preq.get(url).then(function(callRes) {
+			var chtml = cheerio.load(callRes.body);
+
+			return meta.parseEprints(chtml)
+			.then(function(results) {
+				['eprintid', 'datestamp', 'title', 'abstract', 'issn', 'creators_name', 'publication', 'citation'].forEach(function(key) {
+					if(!results[key]) {
+						throw new Error('Expected to find the ' + key + ' key in the response!');
+					}
+				});
+			});
+		});
+	});
+
+	it('should get EPrints author tags correctly', function() {
```

Comment:
I've never tried that so I don't know if it would work- if you did that
then you should definitely wrap the whole thing in a describe function.

The easiest thing to do is to have only one it block and then put multiple
asserts inside the preq block using the same response.
On Dec 30, 2015 9:38 AM, "Scimonster" <notifications@github.com> wrote:

> In test/scraping.js
> <https://github.com/wikimedia/html-metadata/pull/32#discussion_r48594501>:
>
> > +		url = 'http://eprints.gla.ac.uk/113711/';
> > +		return preq.get(url).then(function(callRes) {
> > +			var chtml = cheerio.load(callRes.body);
> > +
> > +			return meta.parseEprints(chtml)
> > +			.then(function(results) {
> > +				['eprintid', 'datestamp', 'title', 'abstract', 'issn', 'creators_name', 'publication', 'citation'].forEach(function(key) {
> > +					if(!results[key]) {
> > +						throw new Error('Expected to find the ' + key + ' key in the response!');
> > +					}
> > +				});
> > +			});
> > +		});
> > +	});
> > +
> > +	it('should get EPrints author tags correctly', function() {
>
> How would i do that? Put both it() tests inside the preq() callback?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/wikimedia/html-metadata/pull/32/files#r48594501>.
>